odoo/addons/base/models/ir_rule.py

```diff
@@ -32,6 +32,16 @@ class IrRule(models.Model):
          'Rule must have at least one checked access right!'),
     ]
 
+    @api.constrains('domain_force')
+    def _check_domain_force(self):
+        eval_context = self._eval_context()
+        for rule in self.filtered('domain_force'):
+            try:
+                dom = safe_eval(rule.domain_force, eval_context)
+                expression.normalize_domain(dom)
```

Comment:
I was thinking why these lines are different from similar code in `_compute_domain`, but then I realized that records are already filtered by `domain_force`
 
https://github.com/odoo/odoo/blob/2b9587aad692b7efc818fc0e66eaf7a3d5956572/odoo/addons/base/models/ir_rule.py#L140-L141